runbotics/runbotics-orchestrator-ui/src/src-app/views/process/ProcessCollectionView/ProcessCollectionModifyDialog/LocationOptions/LocationOptions.tsx

```diff
@@ -0,0 +1,74 @@
+import React, { useEffect, FC } from 'react';
+
+import HomeOutlinedIcon from '@mui/icons-material/HomeOutlined';
+import { Typography, Box, LinearProgress } from '@mui/material';
+import { CollectionId } from 'runbotics-common';
+
+import TreeStructure from '#src-app/components/TreeStructure';
+import { TreeStructureItem } from '#src-app/components/TreeStructure/TreeStructure.types';
+import If from '#src-app/components/utils/If';
+import useTranslations from '#src-app/hooks/useTranslations';
+import { useDispatch, useSelector } from '#src-app/store';
+import { processCollectionActions } from '#src-app/store/slices/ProcessCollection';
+import InfoButtonTooltip from '#src-app/views/process/ProcessBuildView/Modeler/ActionFormPanel/widgets/InfoTooltip/InfoButtonTooltip';
+
+import { LocationOptionsProps } from './LocationOptions.types';
+import { ABSTRACT_ROOT_COLLECTION_ID, accessTooltipIcon, getHierarchicalStructure } from './LocationOptions.utils';
+import { ROOT_PROCESS_COLLECTION_ID } from '../../ProcessCollection.utils';
+
+const LocationOptions: FC<LocationOptionsProps> = ({ isOpen, handleChange, parentId = null }) => {
+    const { translate } = useTranslations();
+    const dispatch = useDispatch();
+    // const { allUserAccessible: { isLoading, list: allUserAccessible } } = useSelector(state => state.processCollection);
+
+    const { allUserAccessible: { isLoading, list: allUserAccessible } } = useSelector(state => state.processCollection);
+    const rootCollections = allUserAccessible.filter(collection => collection.parentId === null);
+    const userAccessibleHierarchy = rootCollections.map((node: any) => getHierarchicalStructure(node, allUserAccessible));
+
+    const rootProcessCollection: TreeStructureItem = {
+        id: ABSTRACT_ROOT_COLLECTION_ID,
+        name: translate('Process.Collection.Structure.Root'),
+        parentId: null,
+        children: userAccessibleHierarchy,
+        icon: accessTooltipIcon.home,
+    };
+
+    const setSelected = (selected: CollectionId[] | CollectionId) => {
+        if (typeof selected === 'string') {
+            if (selected === ABSTRACT_ROOT_COLLECTION_ID) handleChange('parentId', ROOT_PROCESS_COLLECTION_ID);
+            else handleChange('parentId', selected);
+        }
+    };
+
+    useEffect(() => {
+        dispatch(processCollectionActions.getAllAccessible());
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [isOpen]);
```

Comment:
are you sure that you want to fetch it every time? IMO it should be done just on opening